File: 60-permutation-sequence/60-permutation-sequence.java
class Solution {
    public String getPermutation(int n, int k) {
        int fact = 1;
        ArrayList<Integer> list = new ArrayList<>();
        for(int i = 1; i <= n; i++) {
            fact *= i;
            list.add(i);
        }
        
        fact /= n;
        k = k - 1;
        
        StringBuilder ans = new StringBuilder("");
        
        while(true) {
            ans.append(list.get(k / fact));
            list.remove(k / fact);
            if(list.size() == 0) break;
            k = k % fact;
            fact /= list.size();
        }
        
        return ans.toString();
    }
}

File: 60-permutation-sequence/60-permutation-sequence.cpp
class Solution {
public:
    string getPermutation(int n, int k) {
        string nums = "";
        for(int i = 1; i <= n; i++) nums += to_string(i);
        
        string ans = "";
        k--;
        
        while(nums.size() > 0) {
            
            int fact = 1;
            for(int i = 1; i <= nums.size() - 1; i++) fact *= i;
            
            int q = k / fact;
            int r = k % fact;
            
            ans += nums[q];
            k = r;
            
            string temp = "";
            for(int i = 0; i < nums.size(); i++)
                if(i != q) temp += nums[i];
            
            nums = temp;
        }
        
        return ans;
    }
};